clientconn.go

```diff
@@ -1031,6 +1048,13 @@ func (cc *ClientConn) ResetConnectBackoff() {
 func (cc *ClientConn) Close() error {
 	defer cc.cancel()
 
+	cc.csMgr.mu.Lock()
+	if cc.csMgr.connectivityStateTracker != nil {
+		cc.csMgr.connectivityStateTracker.Stop()
+		cc.csMgr.connectivityStateTracker = nil
+	}
+	cc.csMgr.mu.Unlock()
+
```

Comment:
I've moved this code block here, added Lock() and changed to release resources of `cc.csMgr.connectivityStateTracker`.
It would be better to invoke this code block before  `cc.csMgr.updateState(connectivity.Shutdown)` on `Close` method because it wasn't stable whether connectivity shutdown state change would be reported to `Watcher`.